model/task_lifecycle.go

```diff
@@ -609,22 +609,51 @@ func MarkTaskDispatched(t *task.Task, hostId, distroId string) error {
 	return nil
 }
 
+type RestartTaskOptions struct {
+	DryRun     bool
+	OnlyRed    bool
+	OnlyPurple bool
+}
+
+type RestartTaskResults struct {
+	TasksRestarted []string
+	TasksErrored   []string
+}
+
 // RestartFailedTasks attempts to restart failed tasks that started between 2 times
 // It returns a slice of task IDs that were successfully restarted as well as a slice
 // of task IDs that failed to restart
-func RestartFailedTasks(startTime, endTime time.Time, user string, dryRun bool) ([]string, []string, error) {
+// the dryRun parameter will return the tasks that will be restarted if sent true
+// the red and purple params will only restart tasks that were failed due to the test
+// or due to the system, respectively
+func RestartFailedTasks(startTime, endTime time.Time, user string, opts RestartTaskOptions) (RestartTaskResults, error) {
```

Comment:
does it make sense to put more of these options in opts? 

should we write a validation method on opts?

Reply:
I don't think so - none of the other params are optional. I don't know what I would validate since all combinations of opts are valid